src/gsy_e/models/strategy/energy_parameters/heatpump/heat_pump.py

```diff
@@ -294,20 +306,17 @@ def _calc_cop(self, time_slot: DateTime) -> float:
         Generally, the higher the temperature difference between the source and the sink,
         the lower the efficiency of the heat pump (the lower COP).
         """
-        return self._cop_model(
-            self._state.tanks.get_average_tank_temperature(time_slot),
-            self._source_temp_C.get_value(time_slot),
+        # 1 J = 1 W s
+        heat_demand_kW = (
+            self._heat_demand_Q_J.get_value(time_slot) / self._slot_length.total_seconds() / 1000
+            if self._heat_demand_Q_J
+            else None
+        )
+        return self._cop_model.calc_cop(
+            source_temp_C=self._source_temp_C.get_value(time_slot),
+            tank_temp_C=self._state.tanks.get_average_tank_temperature(time_slot),
+            heat_demand_kW=heat_demand_kW,
```

Comment:
Nothing to do here, I took a long time reviewing whether this will break the existing strategies, but as far as I saw the old strategies should operate like before so all good!